public/app/features/dashboard-scene/settings/variables/editors/GroupByVariableEditor.tsx

```diff
@@ -1,12 +1,51 @@
 import React from 'react';
+import { useAsync } from 'react-use';
 
+import { DataSourceInstanceSettings, DataSourceRef, MetricFindValue } from '@grafana/data';
+import { getDataSourceSrv } from '@grafana/runtime';
 import { GroupByVariable } from '@grafana/scenes';
 
+import { GroupByVariableForm } from '../components/GroupByVariableForm';
+
 interface GroupByVariableEditorProps {
   variable: GroupByVariable;
-  onChange: (variable: GroupByVariable) => void;
+  onRunQuery: () => void;
 }
 
 export function GroupByVariableEditor(props: GroupByVariableEditorProps) {
-  return <div>GroupByVariableEditor</div>;
+  const { variable, onRunQuery } = props;
+  const { datasource: datasourceRef, defaultOptions } = variable.useState();
+
+  const { value: datasource } = useAsync(async () => {
+    return await getDataSourceSrv().get(datasourceRef);
+  }, [variable.state]);
+
+  const message = datasource?.getTagKeys
+    ? 'Group by dimensions are applied automatically to all queries that target this data source'
+    : 'This data source does not support group by variable yet.';
+
+  const onDataSourceChange = async (ds: DataSourceInstanceSettings) => {
+    const dsRef: DataSourceRef = {
+      uid: ds.uid,
+      type: ds.type,
+    };
+
+    variable.setState({ datasource: dsRef });
+    onRunQuery();
+  };
+
+  const onDefaultOptionsChange = async (defaultOptions?: MetricFindValue[]) => {
+    variable.setState({ defaultOptions });
+    onRunQuery();
+  };
+
+  return (
+    <GroupByVariableForm
+      defaultOptions={defaultOptions}
+      datasource={datasourceRef ?? undefined}
```

Comment:
I updated here to pass the ref and not the settings, otherwise it will resolve DS variables